test/libyul/objectCompiler/nested_subobject_data_access.yul

```diff
@@ -0,0 +1,92 @@
+object "A" {
+    code {
+        sstore(0, datasize("B.C"))
+        sstore(1, datasize("B.C.dataC1"))
+        sstore(2, datasize("B.C.dataC2"))
+    }
+
+    data "dataA1" "abcdef"
+    data "dataA2" "123"
+
+    object "B" {
+        code {
+            sstore(0, datasize("B"))
+            sstore(1, datasize("dataB1"))
+            sstore(2, datasize("dataB2"))
+        }
+
+        data "dataB1" "abcdef"
+        data "dataB2" "123"
+
+        object "C" {
+            code {
+                sstore(0, datasize("dataC1"))
```

Comment:
Would not hurt to have all paths here for easy comparison with `A`. They should resolve to the same sizes in `A` and `C`.
```suggestion
                sstore(0, datasize("C"))
                sstore(0, datasize("dataC1"))
                sstore(0, datasize("dataC2"))
```